Firestore/Source/Auth/FSTCredentialsProvider.mm

```diff
@@ -121,6 +121,10 @@ - (void)getTokenForcingRefresh:(BOOL)forceRefresh
                                                    userInfo:errorInfo];
             completion(Token::Invalid(), cancelError);
           } else {
+            if (!token) {
+              // We use "" to represent the lack of a token in the C++ Token object.
+              token = @"";
```

Comment:
string_view can be initialized with nullptr, in which case, the .data() returns nullptr, which is what the prepareHeadersForRPC relies on whether to prepare the header or fallback to default. (https://github.com/firebase/firebase-ios-sdk/blob/b7750b588c1d7ae9ea3891a254a39de5d3b3c572/Firestore/Source/Remote/FSTDatastore.mm#L347)

i.e. token string_view can differentiate between nil and "". For nil, rpc.oauth2AccessToken is nil, which makes it fallback to default (without actually write that header line) v.s. for "", which makes it write a header line with token being empty string. This also makes me think we should never allow token being empty string, since it will cause the header being invalid to the server.